.github/workflows/ci.yml

```
@@ -41,3 +42,52 @@ jobs:
+        uses: docker/setup-buildx-action@v3
+
+      - name: Cache Docker layers
+        uses: actions/cache@v4
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.event.pull_request.head.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: 'Login to GitHub Container Registry'
+        uses: docker/login-action@v1
```

actionlint (1.7.11)

[error] the runner of "docker/login-action@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)